controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -593,16 +548,12 @@ public OperationContext createContext(String scope, String name) {
                                                           final UUID txId, final OperationContext context,
                                                           final Executor executor) {
         Stream stream = getStream(scope, streamName, context);
-        CompletableFuture<TxnStatus> future = withCompletion(stream.commitTransaction(txId), executor);
-
-        future.thenCompose(result -> {
-            return stream.getNumberOfOngoingTransactions().thenAccept(count -> {
-                DYNAMIC_LOGGER.incCounterValue(nameFromStream(COMMIT_TRANSACTION, scope, streamName), 1);
-                DYNAMIC_LOGGER.reportGaugeValue(nameFromStream(OPEN_TRANSACTIONS, scope, streamName), count);
-            });
-        });
-
-        return future;
+        return withCompletion(stream.commitTransaction(txId), executor)
+                .thenApply(result -> {
+                    stream.getNumberOfOngoingTransactions().thenAccept(count ->
```

Comment:
this is actually an expensive call --> getNumberOfOngoingTransactions from the store.
should we reassess if we need to have this processing happen at all?

Reply:
Well, a possible way of getting rid of this is by changing `OPEN_TRANSACTIONS` of a Stream from Gauge to Counter. Every time a `createTransaction` is successful on a Stream, we increment this by 1. Every time a `commitTransaction` or `abortTransaction` is successful, we increment this by -1. What do you think?

Reply:
The problem is create stream can happen on any controller instance and hence we can’t do counting